File: backend/src/main/java/com/happyplants/model/dto/PerenualPlantDTO.java
package com.happyplants.model.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public record PerenualPlantDTO(
       @JsonProperty("id")
       Integer perenualId,

        @JsonProperty("common_name")
        String commonName,

        @JsonProperty("scientific_name")
       List<String> scientificName,

        @JsonProperty("family")
        String familyName,

        String cultivar,

        @JsonProperty("species_epithet")
        String speciesEpithet,

        String genus,

        @JsonProperty("watering")
        String wateringDescription,

        @JsonProperty("sunlight")
        String sunDescription
){}

File: frontend/src/types/PerenualPlantDTO.ts
export interface PerenualPlantDTO {
  id?: number;
  common_name?: string;
  scientific_name?: Array<string>;
  family?: string;
  cultivar?: string;
  species_epithet?: string;
  genus?: string;
  description?: string;
  watering?: string;
  sunlight?: Array<string>;
}
